modules/registry/registry-server/registry-api-service/src/main/java/org/apache/airavata/registry/api/service/handler/RegistryServerHandler.java

```diff
@@ -1626,8 +1627,7 @@ public boolean deleteComputeResource(String computeResourceId) throws RegistrySe
     @Override
     public StorageResourceDescription getStorageResource(String storageResourceId) throws RegistryServiceException, TException {
         try {
-            appCatalog = RegistryFactory.getAppCatalog();
-            StorageResourceDescription storageResource = appCatalog.getStorageResource().getStorageResource(storageResourceId);
+            StorageResourceDescription storageResource = new StorageResourceRepository().getStorageResource(storageResourceId);
```

Comment:
Can you create the StorageResourceRepository once for the RegistryServerHandler? Something like this: https://github.com/apache/airavata/pull/185/commits/9292821d004de823222e0edebd8b485160ab2667#diff-930789311194ec4c5b61df3b5d247c59R87

Reply:
Done!. I have to do this for other repository classes as well (GatewayRepository, ComputeRepository etc.)

I will make these changes in another PR.